source/plugins/go/src/oms.go

```diff
@@ -1116,6 +1122,62 @@ func UpdateNumTelegrafMetricsSentTelemetry(numMetricsSent int, numSendErrors int
 	ContainerLogTelemetryMutex.Unlock()
 }
 
+func processIncludes(kubernetesMetadataMap map[string]interface{}, includesList []string) map[string]interface{} {
+	includedMetadata := make(map[string]interface{})
+	for _, include := range includesList {
+		switch include {
+		case "podUid":
+			if val, ok := kubernetesMetadataMap["pod_id"]; ok {
+				includedMetadata["podUid"] = val
+			}
+		case "podLabels":
+			if val, ok := kubernetesMetadataMap["labels"]; ok {
+				includedMetadata["podLabels"] = val
+			}
+		case "podAnnotations":
+			if val, ok := kubernetesMetadataMap["annotations"]; ok {
+				includedMetadata["podAnnotations"] = val
+			}
+		case "image":
+			if hash, ok := kubernetesMetadataMap["container_hash"]; ok {
+				includedMetadata["image_hash"] = hash
+			}
+			if image, ok := kubernetesMetadataMap["container_image"]; ok {
+				includedMetadata["image"] = image
+			}
+		}
+	}
+	return includedMetadata
+}
+
+func convertKubernetesMetadata(kubernetesMetadataJson interface{}) (map[string]interface{}, error) {
```

Comment:
why do we need this convertKubernetesMetadata function? cant we use json.Unmarshal function to convert the JSON string to struct?

Reply:
because this kind of map structure is not directly compatible with the standard way JSON is unmarshalled into Go structs or maps (which typically use map[string]interface{}). So we need to convert it ourselves.